src/api/PDFDocument.ts

```diff
@@ -959,6 +972,125 @@ export default class PDFDocument {
+    const EmbeddedFiles = Names.lookup(PDFName.of('EmbeddedFiles'), PDFDict);
+
+    if (!EmbeddedFiles.has(PDFName.of('Names'))) return [];
+    const EFNames = EmbeddedFiles.lookup(PDFName.of('Names'), PDFArray);
+
+    const rawAttachments = [];
+    for (let idx = 0, len = EFNames.size(); idx < len; idx += 2) {
+      const fileName = EFNames.lookup(idx) as PDFHexString | PDFString;
+      const fileSpec = EFNames.lookup(idx + 1, PDFDict);
+      rawAttachments.push({ fileName, fileSpec });
+    }
+
+    return rawAttachments;
+  }
+
+  private getSavedAttachments(): PDFAttachment[] {
+    const rawAttachments = this.getRawAttachments();
+    return rawAttachments.flatMap(({ fileName, fileSpec }) => {
+      const efDict = fileSpec.lookup(PDFName.of('EF'));
+      if (!(efDict instanceof PDFDict)) return [];
+
+      const stream = efDict.lookup(PDFName.of('F'));
+      if (!(stream instanceof PDFStream)) return [];
+
+      const afr = fileSpec.lookup(PDFName.of('AFRelationship'));
+      const afRelationship =
+        afr instanceof PDFName
+          ? afr.toString().slice(1) // Remove leading slash
+          : afr instanceof PDFString
+            ? afr.decodeText()
+            : undefined;
+
+      const embeddedFileDict = stream.dict;
+      const subtype = embeddedFileDict.lookup(PDFName.of('Subtype'));
+
+      const mimeType =
+        subtype instanceof PDFName
+          ? subtype.toString().slice(1)
+          : subtype instanceof PDFString
+            ? subtype.decodeText()
+            : undefined;
+
+      const paramsDict = embeddedFileDict.lookup(PDFName.of('Params'), PDFDict);
+
+      let creationDate: Date | undefined;
+      let modificationDate: Date | undefined;
+
+      if (paramsDict instanceof PDFDict) {
+        const creationDateRaw = paramsDict.lookup(PDFName.of('CreationDate'));
+        const modDateRaw = paramsDict.lookup(PDFName.of('ModDate'));
+
+        if (creationDateRaw instanceof PDFString) {
+          creationDate = creationDateRaw.decodeDate();
+        }
+
+        if (modDateRaw instanceof PDFString) {
+          modificationDate = modDateRaw.decodeDate();
+        }
+      }
+
+      const description = (
+        fileSpec.lookup(PDFName.of('Desc')) as PDFHexString
+      ).decodeText();
+
+      return [
+        {
+          name: fileName.decodeText(),
+          data: decodePDFRawStream(stream as PDFRawStream).decode(),
+          mimeType: mimeType?.replace(/#([0-9A-Fa-f]{2})/g, (_, hex) =>
+            String.fromCharCode(parseInt(hex, 16)),
+          ),
+          afRelationship: afRelationship as AFRelationship,
+          description,
+          creationDate,
+          modificationDate,
+        },
+      ];
+    });
+  }
+
+  private getUnsavedAttachments(): PDFAttachment[] {
+    type ExposedEmbeddedFile = {
+      embedder: FileEmbedder;
+    };
+
+    const attachments = this.embeddedFiles.map((file) => {
+      const { embedder } = file as unknown as ExposedEmbeddedFile;
```

Comment:
Let's not disable the safeguards. This won't trigger any warning if the implementation changes. You should either make embedder non private or export a method getEmbedder that would return this.embedder. The later is probably a bit better.